### DISCUSSION TITLE
Modifying the generated binary file with pyinstaller

### DISCUSSION THREAD

**Discussion by parsampsh** [pyinstaller/pyinstaller#5443] [Help] [answered]
Hi there! I'm working on a project that is compiled by pyinstaller. so, in a part, in need to change the program binary and generate new output. (in this scenario, source code of program is not accessible). and i ONLY need to change value of a variable and generate the new elf binary with the changed variable. Can you help me? is this possible?

**Comment by rokm** [comment, marked as answer]
What you can do is extract the bytecode-compiled (.pyc) version of the frozen script and the modules it uses. Reversing the process, i.e., mimicking the PyInstaller's assembly pipeline to put it all back together into executable is also doable. 

So I guess the question becomes whether you have the means of editing the bytecode-compiled .pyc file(s) to find and change the variable you are interested in (and whether you know if it's located in the frozen script or in one of the modules, etc.). 

Although you might also be able to decompile the pyc file(s), make the changes, and compile it (them) back. After all, the compilation of python script and modules is done using python's built-in facilities...

**Reply by parsampsh** [reply]
Thank you for your help.
